src/com/cft/contactmerge/NameMatchSelector.java

```diff
@@ -0,0 +1,75 @@
+
+import java.util.ArrayList;
+
+public abstract class NameMatchSelector extends NameMatchLogic {
+    /************************************************************************
+     *********** Method(s) used to call specific name comparisons ***********
+     ************************************************************************/
+    public static AnswerType doFirstNamesMatchUsingAllComparisonChecks(String contactOneFirstName, String contactTwoFirstName){
+        // edit names to go through simple matching algorithms
+        ArrayList<String> editedFirstNameOne = editNameForMatching(contactOneFirstName);
+        ArrayList<String> editedFirstNameTwo = editNameForMatching(contactTwoFirstName);
+
+        return FirstNameMatcher.getFirstNameMatchResult(editedFirstNameOne, editedFirstNameTwo);
+    }
+
+    public static AnswerType doLastNamesMatchUsingAllComparisonChecks(String contactOneLastName, String contactTwoLastName){
+        ArrayList<String> editedLastNameOne = editNameForMatching(contactOneLastName);
+        ArrayList<String> editedLastNameTwo = editNameForMatching(contactTwoLastName);
+
+        return LastNameMatcher.getLastNameMatchResult(editedLastNameOne, editedLastNameTwo);
+    }
+
+    public static AnswerType doFirstAndLastNamesMatchUsingAllComparisonChecks(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
+
+        // edit names to go through simple matching algorithms
+        ArrayList<String> editedFirstNameOne = editNameForMatching(contactOneFirstName);
+        ArrayList<String> editedLastNameOne = editNameForMatching(contactOneLastName);
+        ArrayList<String> editedFirstNameTwo = editNameForMatching(contactTwoFirstName);
+        ArrayList<String> editedLastNameTwo = editNameForMatching(contactTwoLastName);
+
+        // ToDO: Discussion - What if the person got married/divorced and the last name was changed, but first name is still the same?? How to handle this case?
+        // Note: Since first names are more common than last names, lets run last names match first.
+        //       If they don't match, then we don't need to see if first names match, and can skip the first names completely during the matching calculations completely.
+        AnswerType lNameMatchResults = getLastNameMatchResultWithFirstNameSwap(editedFirstNameOne, editedLastNameOne, editedFirstNameTwo, editedLastNameTwo);
+        if (lNameMatchResults.equals(AnswerType.no)){
+            return AnswerType.no;
+        }
+
+        AnswerType fNameMatchResults = getFirstNameMatchResultWithLastNameSwap(editedFirstNameOne, editedLastNameOne, editedFirstNameTwo, editedLastNameTwo);
+        if (fNameMatchResults.equals(AnswerType.no)){
+            return AnswerType.no;
+        }
+
+        if (fNameMatchResults.equals(AnswerType.maybe) || lNameMatchResults.equals(AnswerType.maybe)){
+            return AnswerType.maybe;
+        }
+
+        return AnswerType.yes;
```

Comment:
I agree, but if we make sure no is checked first, the 2nd name won't need to be checked. What this does is increase the speed performance by skipping a bunch of match checks that it doesn't need to. In the first part of this method, no is the default from the return of both first and last name match checks. So if the default is no, return no.

Reply:
We can do both if you are sure the optimization is needed. Keep the early checks but at the end check for cases that are AnswerType.yes and if they don't match then default to no.

Reply:
The problem I see, thinking from a logical point... If a no is found, it can skip the rest of any calculation checking. If a yes/maybe is found, it still has to check the other name in the calculation (adding a cost in performance speed). Lets discuss a good way to reverse this in the next scrum.

Reply:
Nevermind, I was able to find a way to keep the logic while using no, as the default. I will include this update in the next pull.